packages/jsActions/nanoflow-actions-native/src/permissions/RequestGenericPermission.ts

```diff
@@ -0,0 +1,65 @@
+// This file was generated by Mendix Studio Pro.
+//
+// WARNING: Only the following code will be retained when actions are regenerated:
+// - the import list
+// - the code between BEGIN USER CODE and END USER CODE
+// - the code between BEGIN EXTRA CODE and END EXTRA CODE
+// Other code you write will be lost the next time you deploy the project.
+import { Alert, Platform } from "react-native";
+import { check, request, RESULTS, openSettings, PERMISSIONS, Permission } from "react-native-permissions";
+import { ANDROIDPermissionName, IOSPermissionName } from "../../typings/RequestGenericPermission";
+
+// BEGIN EXTRA CODE
+
+function handleBlockedPermission(): void {
+    Alert.alert("title", "need to go to settings for activation", [
+        // draft text
+        { text: "go to settings", onPress: () => openSettings() },
+        { text: "cancel" }
+    ]);
+}
+
+function mapPermissionName(permissionName: string): Permission | undefined {
+    if (Platform.OS === "ios") {
+        const nameWithoutSuffix = permissionName.replace("_IOS", "") as IOSPermissionName;
+
+        return PERMISSIONS.IOS[nameWithoutSuffix];
+    } else if (Platform.OS === "android") {
+        const nameWithoutSuffix = permissionName.replace("_ANDROID", "") as ANDROIDPermissionName;
+
+        return PERMISSIONS.ANDROID[nameWithoutSuffix];
+    }
+}
+// END EXTRA CODE
+
+/**
+ * @param {"NanoflowCommons.Enum_Permissions.CAMERA"|"NanoflowCommons.Enum_Permissions.CONTACTS"} permission - This field is required.
```

Comment:
Shouldn't this be an enum with all possible permissions?

Reply:
done